lib/rpmrc.c

```diff
@@ -453,6 +453,44 @@ const char * lookupInDefaultTable(const char * name,
     return name;
 }
 
+#ifdef MACROFILES
+static char *initMacroPath(const char *confdir)
+{
+    return xstrdup(MACROFILES);
+}
+#else
+/*
+ * Prefer XDG_CONFIG_HOME/rpmmacros but fall back to ~/.rpmmacros
+ * if it exists and the XDG path doesn't.
+ */
+static char *initMacroPath(const char *confdir)
+{
+    const char *xdgconf = getenv("XDG_CONFIG_HOME");
+    if (!(xdgconf && *xdgconf))
+	xdgconf = "~/.config";
+    char *dotpath = rpmGetPath(xdgconf, "/rpmmacros", NULL);
+
+    if (rpmGlob(dotpath, NULL, NULL)) {
+	const char *oldcfg = "~/.rpmmacros";
+	if (rpmGlob(oldcfg, NULL, NULL) == 0) {
```

Comment:
Cosmetic: This second `rpmGlob()` check doesn't really achieve anything; if `~/.rpmmacros` is missing, we'll just keep (the also missing) `~/.config/rpmmacros` in `dotpath`. When later loading the macro files in `rpmInitMacros()`, we'll just skip over missing ones anyway.

Note that if you remove this check, though, you'll need to update the test covering the `XDG_CONFIG_HOME` use case (to actually create the `~/.zzz/rpmconfig` file).

Reply:
The second glob is a bit subtle, but it's quite intentional as per the commit message (emphasis added): "fall back to traditional ~/.rpmmacros *iff* it exists and there's no config in the XDG location."

Without the second glob, we'd advertise the old location in macro path if no per-user configuration is found, whereas we want to advertise the new location going forward. I can clarify this in the commi tmessage though.

Reply:
Oh! This is a good point and something that completely eluded me. Thanks for the explanation!

Reply:
It's also a good reminder of: document not only what, but also *why* - the latter was missing in the original commit message.